projects/pgai/pgai/semantic_catalog/render.py

```diff
@@ -0,0 +1,234 @@
+                    Literal(description.objnames[2]),
+                    Literal(description.description),
+                    Literal(catalog_name),
+                )
+                .as_string(con)
+            )
+        case "view column":
+            assert len(description.objnames) == 3
+            return (
+                SQL("select ai.sc_set_view_col_desc({}, {}, {}, {}, {}, {}, {}, {});\n")  # noqa
+                .format(
+                    Literal(description.classid),
+                    Literal(description.objid),
+                    Literal(description.objsubid),
+                    Literal(description.objnames[0]),
+                    Literal(description.objnames[1]),
+                    Literal(description.objnames[2]),
+                    Literal(description.description),
+                    Literal(catalog_name),
+                )
+                .as_string(con)
+            )
+        case "procedure":
+            assert len(description.objnames) >= 2
+            return (
+                SQL("select ai.sc_set_proc_desc({}, {}, {}, {}, {}, {}, {});\n")
+                .format(
+                    Literal(description.classid),
+                    Literal(description.objid),
+                    Literal(description.objnames[0]),
+                    Literal(description.objnames[1]),
+                    Literal(description.objargs),
+                    Literal(description.description),
+                    Literal(catalog_name),
+                )
+                .as_string(con)
+            )
+        case "function":
+            assert len(description.objnames) >= 2
+            return (
+                SQL("select ai.sc_set_func_desc({}, {}, {}, {}, {}, {}, {});\n")
+                .format(
+                    Literal(description.classid),
+                    Literal(description.objid),
+                    Literal(description.objnames[0]),
+                    Literal(description.objnames[1]),
+                    Literal(description.objargs),
+                    Literal(description.description),
+                    Literal(catalog_name),
+                )
+                .as_string(con)
+            )
+        case "aggregate":
+            assert len(description.objnames) >= 2
+            return (
+                SQL("select ai.sc_set_agg_desc({}, {}, {}, {}, {}, {}, {});\n")
+                .format(
+                    Literal(description.classid),
+                    Literal(description.objid),
+                    Literal(description.objnames[0]),
+                    Literal(description.objnames[1]),
+                    Literal(description.objargs),
+                    Literal(description.description),
+                    Literal(catalog_name),
+                )
+                .as_string(con)
+            )
+        case _:
+            raise ValueError(f"unknown description objtype: {description.objtype}")
+
+
+def render_description_to_comment(
```

Comment:
how is this ever used. How does it get vectorized I mean?

Reply:
It doesn't. I figured if people are generating descriptions automatically, they may also want to apply them as COMMENTs to their database as well. It was nearly a freebee to implement.

Does that answer the question?